pkg/kubelet/dockertools/exec.go

```diff
@@ -90,15 +88,22 @@ func (*NsenterExecHandler) ExecInContainer(client DockerInterface, container *do
 		if stderr != nil {
 			command.Stderr = stderr
 		}
-
-		return command.Run()
+		if err := command.Start(); err != nil {
+			return err
+		}
+	}
+	if timeout > 0 {
+		defer time.AfterFunc(timeout, func() {
+			command.Process.Kill()
+		}).Stop()
```

Comment:
Deferring a call to `Stop()` will clean up resources in case the function returns before the timeout.